src/composables/useFlawCommentsModel.ts

```diff
@@ -0,0 +1,107 @@
+import { computed, ref, type Ref } from 'vue';
+import { getJiraComments, postJiraComment } from '@/services/JiraService';
+import type { ZodFlawCommentType, ZodFlawType } from '@/types/zodFlaw';
+import { postFlawComment } from '@/services/FlawService';
+
+export const SYSTEM_EMAIL = 'bugzilla@redhat.com';
+
+export function useFlawCommentsModel(flaw: Ref<ZodFlawType>, isSaving: Ref<boolean>, afterSaveSuccess: () => void) {
+  const internalCommentsAvailable = ref(false);
+  const isLoadingInternalComments = ref(false);
+  const internalComments = ref<ZodFlawCommentType[]>([]);
+
+  const publicComments = computed<ZodFlawCommentType[]>(() => flaw.value.comments.filter(filterFunctions.public));
+  const privateComments = computed<ZodFlawCommentType[]>(() => flaw.value.comments.filter(filterFunctions.private));
+  const systemComments = computed<ZodFlawCommentType[]>(() => flaw.value.comments.filter(filterFunctions.system));
+
+  type OsidbCommentFilter = 'public' | 'private' | 'system';
+  type OsidbCommentFilterFunctions = Record<OsidbCommentFilter, (comment: ZodFlawCommentType) => boolean>;
+
+  const filterFunctions: OsidbCommentFilterFunctions = {
+    public: (comment: any) => !comment.is_private,
+    private: (comment: any) => comment.is_private && comment.creator !== SYSTEM_EMAIL,
+    system: (comment: any) => comment.creator === SYSTEM_EMAIL,
+  };
```

Comment:
This will be recreated each time the `useFlawCommentsModel` function is called, since they don't use any variable from the `useFlawCommentsModel` I think we can move them outside of the body of the function